### PR TITLE
Menu item: allow anchor tags

### PR DESCRIPTION
**Pull request by driesd**
### Description

This PR adds an `element` prop which accepts either a `button` (default value) or an `anchor tag`. I also needed to adjust the markup and styling to make sure it still looks like the intended design.

Also, I made it keyboard accessible:

Screenshot 2019-11-21 10 37 09

### Breaking changes

None.

### CODE REVIEW

**Comment by ArnaudWeyts** [comment]
⏳  Manually checking

File: src/components/menu/MenuItem.js [hunk 1]
```diff
@@ -30,34 +30,42 @@ class MenuItem extends PureComponent {
...
+          textAlign="left"
+          {...(element === 'a' && disabled && { tabIndex: '-1' })}
+          {...restProps}
+        >
+          {icon && (
+            <Icon color={color} tint={tint} marginRight={3}>
+              {icon}
+            </Icon>
           )}
+          <Box flex="1 0 auto">
```
Comment (kevinwaelkens) [inline src/components/menu/MenuItem.js]: Actually we only want to render this Box when there's a label or a caption right? 
```suggestion
          {label || caption (<Box flex="1 0 auto">
```

File: src/components/menu/MenuItem.js [hunk 2]
```diff
@@ -30,34 +30,42 @@ class MenuItem extends PureComponent {
...
+              <TextBody color={color} tint={tint}>
+                {label}
+              </TextBody>
+            )}
+            {caption && (
+              <TextBody color="neutral" tint="darkest">
+                {caption}
+              </TextBody>
+            )}
+          </Box>
```
Comment (kevinwaelkens) [inline src/components/menu/MenuItem.js]:
```suggestion
          </Box>)}
```

File: src/components/menu/MenuItem.js (hunk 1 above)
Comment (ArnaudWeyts) [reply src/components/menu/MenuItem.js]: Isn't the label required?

**Comment by ArnaudWeyts** [comment]
✅ Manually checked

File: src/components/menu/MenuItem.js (hunk 1 above)
Comment (kevinwaelkens) [reply src/components/menu/MenuItem.js]: Doesn't seem required, it's indeed not really a logical use-case, but yeah 🤷‍♂

**Pull request merged**